numpy/core/src/multiarray/datetime_strings.c

```diff
@@ -484,7 +484,7 @@ parse_iso_8601_datetime(char *str, Py_ssize_t len,
     if (sublen >= 2 && isdigit(substr[0]) && isdigit(substr[1])) {
         out->hour = 10 * (substr[0] - '0') + (substr[1] - '0');
```

Comment:
`substr[0]` and `substr[1]` are `char` digits (`isdigit`), of which `'0'` is the smallest, so `out->hour` can only be as small as `10 * 0 + 0 = 0`